### PR TITLE
Increased hooks usage and fix ETH zero and empty wallet related errors

### PR DESCRIPTION
**Pull request by jinchung**
This PR includes moving a lot of hocs to hooks and hookifying some more components.

One issue I would like help with: Switching network back to mainnet has an interaction delay when clicking Done

### CODE REVIEW

File: src/components/header/ProfileHeaderButton.js [hunk 1]
```diff
@@ -32,76 +34,62 @@ const FirstLetter = styled(Text)`
...
+  const { pendingRequestCount } = useRequests();
+  const { isCoinListEdited } = useCoinListEdited();
+  const { accountColor, accountName } = useAccountSettings();
+
+  const onPress = useCallback(
+    () => navigation.navigate(Routes.PROFILE_SCREEN),
+    [navigation]
+  );
 
-ProfileHeaderButton.propTypes = {
```
Comment (brunobar79) [inline src/components/header/ProfileHeaderButton.js:88]: Do you mind keeping the propTypes here?

**Comment by osdnk** [comment]
Can we cherry-pick some commits to `develop`? Reviewing it is quite difficult

File: src/components/send/SendAssetFormToken.js [hunk 2]
```diff
@@ -19,65 +16,65 @@ const SendAssetFormToken = ({
...
-      style={{ zIndex: 3 }}
-      width="100%"
-    >
-      {buttonRenderer}
-      {txSpeedRenderer}
-    </ColumnWithMargins>
-  </>
-);
+}) => {
+  const formatNativeInput = useCallback(
```
Comment (brunobar79) [inline src/components/send/SendAssetFormToken.js:20]: WDYT about moving this to utils?

File: src/screens/SettingsModal.js [hunk 3]
```diff
@@ -54,17 +53,35 @@ const requestFaceIDPermission = () =>
...
+
+  const onPressBack = useCallback(
+    () => navigation.setParams({ section: SettingsPages.default }),
+    [navigation]
+  );
+
+  const onPressImportSeedPhrase = useCallback(() => {
+    InteractionManager.runAfterInteractions(() => {
+      navigation.navigate(Routes.IMPORT_SEED_PHRASE_SHEET);
+      StatusBar.setBarStyle('light-content');
```
Comment (brunobar79) [inline src/screens/SettingsModal.js:76]: I'd keep all the statusbar logic in the same place (Routes.ios | android ) => onNavigationStateChange

File: src/components/send/SendAssetFormToken.js (hunk 2 above)
Comment (jinchung) [reply src/components/send/SendAssetFormToken.js:20]: Will do this separately!

File: src/screens/SettingsModal.js (hunk 3 above)
Comment (jinchung) [reply src/screens/SettingsModal.js:76]: Will do this separately!

**Pull request merged**